### PR TITLE
test: rename slack.unit.test.ts → slack.integration.test.ts + expand to 2437 lines (ATP-3.2)

### PR DESCRIPTION
**Pull request by dmcaulay**
## Summary
- Renames `agent/src/slack.unit.test.ts` → `slack.integration.test.ts` (label correction — file always used injected deps + real file I/O)
- Expands from 407 → 2437 lines: adds voice message/reply paths, `downloadFile` edge cases, `reaction_added` handler coverage, full marker dispatch surface, `formatRunErrorForSlack` edge cases including `ClaudeTimeoutError`, and `voiceConfig` set vs absent scenarios
- Uses vitals-os as source; all tests use injected deps via `createSlackApp` params — no `mock.module()` or global mutations

## Acceptance Criteria
| # | Criterion | Status |
|---|-----------|--------|
| 1 | Old file deleted; new file exists | ✅ MET |
| 2 | Voice message path (transcribeFn call) | ✅ MET |
| 3 | Voice reply path ([speak:text] + uploadV2) | ✅ MET |
| 4 | downloadFile coverage (size/url guards) | ⚠ PARTIAL — HTTP paths require fetchFn injection not added in this task |
| 5 | reaction_added handler coverage | ✅ MET |
| 6 | dispatchMarkers full coverage | ✅ MET |
| 7 | formatRunErrorForSlack edge cases | ✅ MET |
| 8 | voiceConfig set vs absent | ✅ MET |
| 9 | All existing tests preserved | ✅ MET (126 pass) |
| 10 | No test-env.ts import | ✅ MET |
| 11 | bun test --filter agent passes | ✅ MET (516 pass, 0 fail) |

## Test Plan
- [x] `bun test --filter agent` — 516 pass, 0 fail
- [x] `bunx biome lint agent/src/slack.integration.test.ts` — no issues
- [x] `bun run --filter='*' typecheck` — exit 0

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: agent/src/slack.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,2592 @@
...
+ * needed anywhere.
+ *
+ * - appFactory: MockApp class that captures constructor args + handlers
+ * - runner: mock function
+ * - formatter: mock function
+ * - getThreadKey: real threadKey (pure function, no deps)
+ * - slackConfig: plain object with test credentials
+ */
+
+import { afterEach, beforeEach, describe, expect, mock, test } from "bun:test";
```
Comment (dmcaulay) [inline agent/src/slack.integration.test.ts]: `afterEach` is imported here but never called in this file. Remove it to keep the import list clean.

**Review by dmcaulay — COMMENTED** [review]
APPROVE — 126 tests across the full message/mention/reaction/marker/voice/error surface. Correct integration layer (real file I/O for voice and upload tests). AC#4 downloadFile HTTP paths are tracked in ATP-1.5. No mock.module(), no global mutations. CI green.

(Posted as COMMENT — GitHub blocks self-review.)

**Review by dodizzle — APPROVED** [review]
APPROVE — layer label corrected (real file I/O + injected doubles = integration, not unit). 21 describe blocks covering the full handler surface: DM routing, channel thread routing, file handling, app_mention, voice transcription, [speak:text], reaction_added, invalid_blocks retry, formatRunErrorForSlack, thread history injection, and dispatchMarkers. No mock.module(), no global.* mutations, isolation contract fully respected. CI green (516 pass).

One minor suggestion: `test("processes file_share messages")` (line 253) calls `createSlackApp()` twice — once in beforeEach and once in the test body. Works correctly and matches the override pattern used throughout, but a brief inline comment would make the intent explicit.

**Review by dodizzle — APPROVED** [review]
APPROVE — unused `afterEach` import removed (lint cleanup, no behavioral change). All prior findings remain resolved. CI green (516 pass). Previous APPROVE stands.

One open suggestion (non-blocking, from prior review): `agent/src/slack.integration.test.ts:253` — `test("processes file_share messages")` calls `createSlackApp()` twice (once in `beforeEach`, once in the test body for the override). A brief inline comment `// override fileDownloaderFn — reinitializes handler` would make the intent explicit.

File: agent/src/slack.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,2592 @@
...
+    expect(out).toContain("retry");
+    expect(out).not.toContain("```");
+  });
+
+  test("ClaudeRunError with undefined sessionId handled gracefully", () => {
+    const err = new ClaudeRunError("msg", 500, "Internal server error", undefined);
+    const out = formatRunErrorForSlack(err);
+    expect(out).toContain("500");
+    expect(out).not.toThrow;
+  });
```
Comment (dmcaulay) [inline agent/src/slack.integration.test.ts:2178]: `expect(out).not.toThrow` is a property access, not a function call — this assertion passes unconditionally and provides no safety net. Change to `expect(() => formatRunErrorForSlack(err)).not.toThrow()` to actually assert the function does not throw.

**Review by dmcaulay — COMMENTED** [review]
COMMENT — one no-op assertion at line 2178 silently passes without testing anything and must be fixed before merge.

The "ClaudeRunError with undefined sessionId handled gracefully" test contains `expect(out).not.toThrow` — a property access, not a function call. This passes unconditionally regardless of whether `formatRunErrorForSlack` throws. Fix: `expect(() => formatRunErrorForSlack(err)).not.toThrow()`.

Everything else is solid: strict DI via `createSlackApp` params, no `mock.module()` or global mutations, correct integration layer classification, comprehensive surface coverage.

(Posted as COMMENT — GitHub blocks self-review.)

**Review by dodizzle — APPROVED** [review]
APPROVE — prior blocking finding addressed: `expect(out).not.toThrow` (no-op property access) correctly replaced with `expect(() => formatRunErrorForSlack(err)).not.toThrow()`. CI green. No new issues.

**Review by dmcaulay — COMMENTED** [review]
APPROVE — prior blocking finding addressed: `expect(out).not.toThrow` (no-op property access) correctly replaced with `expect(() => formatRunErrorForSlack(err)).not.toThrow()`. CI green. No new issues.

(Posted as COMMENT — GitHub blocks self-review.)
